src/main/scala/com/github.sstone/amqp/ChannelOwner.scala

```diff
@@ -51,17 +57,17 @@ object ChannelOwner {
           def handleFlow(active: Boolean): Unit = listener ! HandleFlow(active)
         }))
       }
-      case request@Publish(exchange, routingKey, body, properties, mandatory, immediate) => {
+      case request: Publish => {
         log.debug("publishing %s".format(request))
-        val props = properties getOrElse new AMQP.BasicProperties.Builder().build()
-        sender ! withChannel(channel, request)(c => c.basicPublish(exchange, routingKey, mandatory, immediate, props, body))
+        sender ! withChannel(channel, request)(c => publishAndReturnSeqNo(c, request))
       }
       case request@Transaction(publish) => {
         sender ! withChannel(channel, request) {
           c => {
             c.txSelect()
-            publish.foreach(p => c.basicPublish(p.exchange, p.key, p.mandatory, p.immediate, p.properties getOrElse new AMQP.BasicProperties.Builder().build(), p.body))
+            val seqNos = publish.map(p => publishAndReturnSeqNo(c, p))
```

Comment:
We don't need this with transactions, because they are blocking until every message is delivered. 
http://www.rabbitmq.com/blog/2011/02/10/introducing-publisher-confirms/